kaupunginnaiset/alman-akka
# Add initial events data and update script

## Purpose

* Refactored folder structure so that db scripts are in tools folder
* Added `data`-folder that contains the service events as json. The events are rendered to app main view (as a first test).
* Added script (`fetch`) for updating the events json from Firebase. The script is run daily via GitHub actions (can be triggered also from the actions UI). Logic for the data updates will be tuned more later.
* The `fetch`-script also generates an `index.js`-file for easily include the events to React-components. Currently it includes all available events, but in production-phase the purpose is to include only the events in current month and onwards. 

## Related Issues

I will create later issues for the missing functionality such as unit tests for the db scripts and issues mentioned above.

## How to test

DB events inclusion to React: Start app in localhost and see that events are listed with index and title to main view.
DB scripts: Open Actions tab and see that the nightly run for workflow `fetch-data` has succeeded.
